tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/observability/metrics/MetricsTestCase.java

```diff
@@ -125,21 +125,41 @@ private Metrics filterByTag(Metrics metrics, String key, String value) {
     /**
      * Test the metrics generated for a particular function invocation.
      *
-     * @param allMetrics All the metrics collected from Metrics Registry
+     * @param allMetrics         All the metrics collected from Metrics Registry
      * @param invocationPosition The invocation position of the function invocation
-     * @param invocationCount The number of times the function should have been called
-     * @param additionalTags Additional tags that should be present in the metrics
+     * @param invocationCount    The number of times the function should have been called
+     * @param additionalTags     Additional tags that should be present in the metrics
+     */
+    private void testFunctionMetrics(Metrics allMetrics, String invocationPosition, long invocationCount,
+                                     Tag... additionalTags) {
+
+        testFunctionMetrics(allMetrics, invocationPosition, invocationCount, additionalTags, null);
+    }
+
+    /**
+     * Test the metrics generated for a particular function invocation for a given start and end tag set.
+     *
+     * @param allMetrics           All the metrics collected from Metrics Registry
+     * @param invocationPosition   The invocation position of the function invocation
+     * @param invocationCount      The number of times the function should have been called
+     * @param startObservationTags tags at the observation start which should be present in the metrics
+     * @param endObservationTags   tags at the observation end which should be present in the metrics
```

Comment:
We should rename the param and rephrase the description. `endObservationTags` is actually the additional tags added after start here.